packages/ts-transformers/src/schema/schema-transformer.ts

```diff
@@ -33,7 +36,16 @@ export function createSchemaTransformer(
         if (!typeArg) {
           return ts.visitEachChild(node, visit, context);
         }
-        const type = checker.getTypeFromTypeNode(typeArg);
+
+        // First check if we have a registered Type for this node
+        // (from schema-injection when synthetic TypeNodes were created)
+        let type: ts.Type;
+        if (typeRegistry && typeRegistry.has(node)) {
```

Comment:
Are these the same Nodes that have registered to the type registry, e.g. the same instance (as opposed to a different instance representing the same underlying syntax fragment)?